Objects/tupleobject.c

```diff
@@ -333,39 +333,49 @@ tuplerepr(PyTupleObject *v)
     return NULL;
 }
 
-/* The addend 82520, was selected from the range(0, 1000000) for
-   generating the greatest number of prime multipliers for tuples
-   up to length eight:
 
-     1082527, 1165049, 1082531, 1165057, 1247581, 1330103, 1082533,
-     1330111, 1412633, 1165069, 1247599, 1495177, 1577699
+/* Hash for tuples. This is a slightly simplified version of the xxHash
+   non-cryptographic hash: we do not use any parallellism, there is only
+   1 accumulator. For the specification, see
+   https://github.com/Cyan4973/xxHash/wiki/xxHash-specification-(draft)
 
    Tests have shown that it's not worth to cache the hash value, see
-   issue #9685.
+   https://bugs.python.org/issue9685
 */
-
 static Py_hash_t
 tuplehash(PyTupleObject *v)
 {
-    Py_uhash_t x;  /* Unsigned for defined overflow behavior. */
-    Py_hash_t y;
-    Py_ssize_t len = Py_SIZE(v);
-    PyObject **p;
-    Py_uhash_t mult = _PyHASH_MULTIPLIER;
-    x = 0x345678UL;
-    p = v->ob_item;
-    while (--len >= 0) {
-        y = PyObject_Hash(*p++);
-        if (y == -1)
+    Py_uhash_t ERR = -1;
```

Comment:
This is akin to, e.g., `int TWO = 2;" :wink:

 `-1` is hardcoded all over almost all API functions that return integers to mean "error.  Giving it a symbolic name instead suggests instead this function uses some other convention - which it doesn't.

Reply:
It's mainly to avoid the case: `ERR` is a lot shorter to write than `(Py_uhash_t)-1`. Of course, this is a pure bikeshedding issue so I'll happily write `(Py_uhash_t)-1` if you insist.

Reply:
Code is read far more often than it's written.  `(Py_uhash_t)-1` is wholly explicit, obvious, and self-contained.